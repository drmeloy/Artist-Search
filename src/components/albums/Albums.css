.Albums {
  z-index: 1;
  position: fixed;
  height: 100vh;
  width: 33vw;
  right: 0;
  top: 0;
  background: black;
  color: white;
  border-left: 3px solid;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: auto;  
  margin-top: 0;
  animation: slidein .5s 1;

  a {
    text-decoration: none;
    color: white;
  }

  ul {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 2px solid;
  }

  li {
    list-style: none;
    margin: 30px;
    text-align: center;
  }

  img {
    width: 15vw;
  }

  a {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  p {
    margin: 10px;
  }

  span {
    font-size: 2rem;
    text-align: center;
  }

  details {
    text-align: center;
    max-width: 80%;
    margin-top: 30px;
  }

  button {
    position: fixed;
    bottom: 2vh;
    left: 69vw;
    color: white;
    background: black;
    font-size: 1rem;
  }

  @keyframes slidein {
    from {
      right: -100vw;
    }

    to {
      right: 0;
    }
  }
}

.SlideOut {
  animation: slideout .5s 1;
  
  @keyframes slideout {
    from {
      right: 0;
    }

    to {
      right: -100vw;
    }
  }
}