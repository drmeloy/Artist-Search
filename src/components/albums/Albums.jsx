import React from 'react';
import PropTypes from 'prop-types';
import { useGetAlbums } from '../hooks/albums';
import styles from './Albums.css';
import SongList from '../songlist/Songlist.jsx';
import { useHistory } from 'react-router-dom';

const Albums = ({ match }) => {  
  const { artistName, artistId } = match.params;
  const history = useHistory();

  const goBack = ({ target }) => {
    target.parentNode.classList.add(styles.SlideOut);
    setTimeout(() => {
      history.replace('');
    }, 500);
  };

  const albums = useGetAlbums(artistId);

  const albumList = albums.map(album => {
    const src = album?.coverArt?.front ? `http://coverartarchive.org/release/${album.id}/front` : 'https://m.mrjatt-mp3.com/cover.jpg';
    const alt = album?.coverArt?.front ? `${album.title} cover art` : 'No cover art available';
    
    return (
      <>
        <details>
          <summary>
            <img src={src} alt={alt} />
            <p>{album.title}</p>
            <p>{album.date}</p>
          </summary>
          <SongList artistName={artistName} artistId={artistId} albumId={album.id} />
        </details>
      </>
    );
  });

  return (
    <>
      <summary className={styles.Albums}>
        <span>Albums by {artistName}</span>
        {albumList}
        <button onClick={goBack}>&#8674;</button>
      </summary>
    </>
  );
};

Albums.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      artistName: PropTypes.string.isRequired,
      artistId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Albums;
