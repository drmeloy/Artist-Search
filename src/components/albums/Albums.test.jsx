import React from 'react';
import { shallow } from 'enzyme';
import Albums from './Albums';

describe('Albums component', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<Albums match={{ params: { artistName: '', artistId: '' }}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
