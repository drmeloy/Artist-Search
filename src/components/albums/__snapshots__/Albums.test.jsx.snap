// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums component matches its snapshot 1`] = `
<ul>
  <li>
    <Link
      to="///undefined"
    >
      <img
        alt="No cover art available"
        src="https://m.mrjatt-mp3.com/cover.jpg"
        style={
          Object {
            "width": "15vw",
          }
        }
      />
      <p />
      <p />
    </Link>
  </li>
</ul>
`;
